crates/hyperqueue/src/server/job.rs

```diff
@@ -357,11 +358,15 @@ impl Job {
             JobTaskState::Running { started_data, .. } => {
                 *state = JobTaskState::Canceled {
                     started_data: Some(started_data.clone()),
+                    cancelled_date: Some(now),
                 };
                 self.counters.n_running_tasks -= 1;
             }
             JobTaskState::Waiting => {
-                *state = JobTaskState::Canceled { started_data: None };
+                *state = JobTaskState::Canceled {
+                    started_data: None,
+                    cancelled_date: None,
```

Comment:
We always know the cancel date, so let's set it here and remove `Option` from `cancelled_date`.